clients/web/src/lib/utils/actions/clickOutsideAction.ts

```diff
@@ -1,34 +1,13 @@
-const ClickOutsideManager = {
-    registeredListeners: new Set<[CallableFunction, HTMLElement]>(),
-    boundFunction: undefined,
-
-    handleOnClick: function(event: PointerEvent) {
-        this.registeredListeners.forEach(([func, el]) => {
-            if (el && !el.contains(event.target) && !event.defaultPrevented) {
-                func();
-            }
-        });
-    },
-    
-    addListener: function(func: CallableFunction, el: HTMLElement) {
-        if (this.registeredListeners.size === 0) {
-            // Start listening to document clicks
-            this.boundFunction = this.handleOnClick.bind(this);
-            document.addEventListener("click", this.boundFunction);
+export function clickOutside(node) {
+    const handleClick = event => {
+        if (!node.contains(event.target)) {
+            node.dispatchEvent(new CustomEvent("outclick"));
```

Comment:
Why are custom events required here?

Reply:
Because the modal uses on:outclick. The customs event is distinguished from a click on the buttons of the modals this way. (At least I think this is how that works) a click is a click otherwise right?

Reply:
You can distinguish where a click happens by using el.contains, similarly to the previous implementation; I don't think we should add a custom event that fires for this